napari/utils/colormaps/colormap.py

```diff
@@ -18,12 +19,11 @@ class ColormapInterpolationMode(StringEnum):
               bin between by neighboring controls points.
     """
 
-    LINEAR = auto()
-    ZERO = auto()
+    LINEAR = 'linear'
+    ZERO = 'zero'
 
 
-@evented_dataclass
-class Colormap:
+class Colormap(EventedModel):
```

Comment:
@sofroniewn , do you think it's worth adding `__eq__` methods so that `cmap_1 == cmap_2` works? This is useful for when an `EventedModel` has a `Colormap` as a field (i.e., for the [equality check for emitting an event when setting the field value](https://github.com/napari/napari/blob/538c1e4fee9bd272ea193511679568bd25117375/napari/utils/events/evented_model.py#L70)).

This might apply for the other models in this PR, but I came across this one while working on the `ColorManager` PR.

Reply:
Yes! It sounds like we need to add this because of the `Array` property in the Colormap. We already have a `__equality_checks__` dictionary, but I think we just need to wire things up properly. I can take a look soon cc @tlambert03

Reply:
Note a PR to address this has been started in #2200